internal/weldr/api.go

```diff
@@ -156,7 +177,8 @@ func New(repoPaths []string, stateDir string, rpm rpmmd.RPMMD, distros *distrore
 		logger:          logger,
 		compatOutputDir: compatOutputDir,
 		hostDistroName:  hostDistro.Name(),
-		distros:         distros,
+		distroRegistry:  dr,
+		distros:         validDistros(rr, dr, hostArch.Name(), logger),
```

Comment:
I like this approach! Allows us to easily select which distros to support where in the .spec file, by just shipping (or not) the repo files. (If I understood correctly).

Reply:
Correct, also see my reply above.